app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsRTDB.kt

```diff
@@ -18,38 +15,47 @@ import java.util.concurrent.CompletableFuture
 class UserCardsRTDB {
     private val db = Firebase.database.reference
     private val user = Firebase.auth.currentUser
-    private val userCardCollection = db.child(user!!.uid)
+    private val userCardCollection = if (user != null) db.child(user.uid) else null
+
+
+    /**
+     * Checks if the user is connected.
+     */
+    fun isConnected(): Boolean {
+        return user != null
+    }
 
     /**
      * Adds a card to the user's collection.
      */
-    fun addCardToCollection(card: MagicCard) {
-        val cardUID = card.set.code + card.number
-        val data = Gson().toJson(card)
-        userCardCollection.child(cardUID).setValue(data)
+    fun addCardToCollection(fbCard: FBMagicCard) {
+        val cardUID = fbCard.card.set.code + fbCard.card.number
+        val data = Gson().toJson(fbCard)
+        userCardCollection?.child(cardUID)?.setValue(data)
     }
 
     /**
      * Removes a card from the user's collection.
      */
-    fun removeCardFromCollection(card: MagicCard) {
-        val cardUID = card.set.code + card.number
-        userCardCollection.child(cardUID).removeValue()
+    fun removeCardFromCollection(fbCard: FBMagicCard) {

```

Comment:
Why using FBMagicCard here? We don't care about the possession of a card.

Reply:
We use FBMagicCard in all the methods of the database, this is to keep the code consistent.